### PR TITLE
Document XML_OPTION_PARSE_HUGE

### PR DESCRIPTION
**Pull request by nielsdos**

### CODE REVIEW

File: reference/xml/functions/xml-parser-set-option.xml [hunk 1]
```diff
@@ -59,6 +59,15 @@
             XML parser.  Enabled by default.
            </entry>
           </row>
+          <row>
+           <entry><constant>XML_OPTION_PARSE_HUGE</constant></entry>
+           <entry>bool</entry>
+           <entry>
+            Allows parsing documents larger than 10 MB.
+            This option should only be enabled when the document size is
+            bounded.
```
Comment (cmb69) [inline reference/xml/functions/xml-parser-set-option.xml:68]: Probably clear enough, but maybe hint at DoS?

File: reference/xml/constants.xml [hunk 2]
```diff
@@ -257,6 +257,17 @@
...
    </listitem>
   </varlistentry>
+  <varlistentry xml:id="constant.xml-option-parse-huge">
+   <term>
+    <constant>XML_OPTION_PARSE_HUGE</constant>
+    (<type>int</type>)
+   </term>
+   <listitem>
+    <simpara>
+     Available as of PHP 8.4.0.
```
Comment (cmb69) [inline reference/xml/constants.xml:267]: I have no idea whether libexpat is still properly supported, but if it is, we should clarify here (or maybe below) that the constant is only relevant for libxml2. I'm also not sure whether PHP 7.0 still supported libxml2 < 2.7.0, and if so, what would happen there (would basically `XML_OPTION_PARSE_HUGE` be enabled by default, and could not be disabled?)

**Review by cmb69 — COMMENTED** [review]
Thank you! Maybe add some clarifications, but generally looks good.

File: reference/xml/constants.xml (hunk 2 above)
Comment (nielsdos) [reply reference/xml/constants.xml:267]: you are right; why does it have to be so complex :(

Comment (nielsdos) [reply reference/xml/constants.xml:267]: Just checked, PHP 7 supports libxml2 2.6.11

**Review by cmb69 — APPROVED** [review]
Thank you!

**Pull request merged**